django/db/models/base.py

```diff
@@ -322,11 +322,23 @@ def __new__(cls, name, bases, attrs, **kwargs):
         return new_class
 
     def add_to_class(cls, name, value):
+        if name in cls.__dict__:  # not hasattr() to avoid parent attrs
+            raise AttributeError(
+                f"Class {cls.__name__} has an attribute '{name}' which would be "
+                f"overridden. Check definitions."
+            )
         if _has_contribute_to_class(value):
             value.contribute_to_class(cls, name)
         else:
             setattr(cls, name, value)
 
+    def create_auto_pk(cls, pk_class):
+        pk_field = pk_class(verbose_name='ID', primary_key=True, auto_created=True)
+        try:
+            cls.add_to_class('id', pk_field)
+        except Exception:
+            raise FieldError(f"Adding a generated Primary Key field to model {cls.__name__} failed.")
```

Comment:
You are shadowing the original exception here, which could be useful for debugging.
I am also not use your exception adds a lot more helpful context. Capturing `Exception` or `BaseException` is usually a red flag, as the underlying error could be anything.

I would suggest removing the whole exception handling part. After that, it stands to reason, if you need to new two line method.

If not, we should at least include the original exception.
```suggestion
        try:
            cls.add_to_class('id', pk_field)
        except Exception as e:
            raise FieldError(f"Adding the automatically gennerated primary key field to {cls.__qualname__} failed.") from e
```